extensions/vertx/deployment/src/test/java/io/quarkus/vertx/deployment/MessageConsumerContextTest.java

```diff
@@ -0,0 +1,161 @@
+package io.quarkus.vertx.deployment;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.time.Duration;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.quarkus.vertx.ConsumeEvent;
+import io.smallrye.common.annotation.Blocking;
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.impl.ConcurrentHashSet;
+import io.vertx.mutiny.core.eventbus.EventBus;
+import io.vertx.mutiny.core.eventbus.Message;
+
+/**
+ * Verify that event consumer are attached to different contexts.
+ */
+public class MessageConsumerContextTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class).addClasses(MessageConsumers.class));
+
+    @Inject
+    MessageConsumers messageConsumers;
+
+    @Inject
+    EventBus eventBus;
+
+    @RepeatedTest(5)
+    public void testSend() throws InterruptedException {
+        MessageConsumers.MESSAGES.clear();
+        MessageConsumers.latch = new CountDownLatch(3);
+        eventBus.send("send", "foo");
+        eventBus.send("send", "bar");
+        eventBus.send("send", "baz");
+        assertTrue(MessageConsumers.latch.await(3, TimeUnit.SECONDS));
+        if (Runtime.getRuntime().availableProcessors() > 1) {
+            assertEquals(3, MessageConsumers.MESSAGES.size());
+        } else {
+            assertTrue(MessageConsumers.MESSAGES.size() >= 2);
+        }
+    }
+
+    @RepeatedTest(5)
+    public void testPublish() throws InterruptedException {
+        MessageConsumers.MESSAGES.clear();
+        MessageConsumers.latch = new CountDownLatch(9); // 3 messages x 3 consumers
+        eventBus.publish("pub", "foo");
+        eventBus.publish("pub", "bar");
+        eventBus.publish("pub", "baz");
+        assertTrue(MessageConsumers.latch.await(3, TimeUnit.SECONDS));
+        if (Runtime.getRuntime().availableProcessors() > 1) {
```

Comment:
Why do we need to consider 2 cases here?

Reply:
Because CI runs with no parallelism and so, we end up with _not enough event loop_.

Reply:
Oh yeah...